src/System.Text.Json/src/System/Text/Json/JsonReaderHelper.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json
+{
+    internal static class JsonReaderHelper
+    {
+        public static (int, int) CountNewLines(ReadOnlySpan<byte> data)
+        {
+            int lastLineFeedIndex = -1;
+            int newLines = 0;
+            for (int i = 0; i < data.Length; i++)
+            {
+                if (data[i] == JsonConstants.LineFeed)
+                {
+                    lastLineFeedIndex = i;
+                    newLines++;
+                }
+            }
+            return (newLines, lastLineFeedIndex);
+        }
+
+        // https://tools.ietf.org/html/rfc8259
+        // Does the span contain '\' or any control characters (i.e. 0 to 31)
+        // IndexOfAny(92, < 32)
+        // Borrowed and modified from SpanHelpers.Byte:
+        // https://github.com/dotnet/corefx/blob/fc169cddedb6820aaabbdb8b7bece2a3df0fd1a5/src/Common/src/CoreLib/System/SpanHelpers.Byte.cs#L473-L604
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int IndexOfAnyControlOrEscape(this ReadOnlySpan<byte> span)
+        {
+            return IndexOfAny(
+                    ref MemoryMarshal.GetReference(span),
+                    JsonConstants.ReverseSolidus,
+                    JsonConstants.Space,
```

Comment:
I'm not clear how this implementation matches with either the method name or the comment.  The comment says it matches 0-31+'/', the name says it matches 0-31+'Esc', and this seems to match 0-31+' '.  Am I misunderstanding?

Reply:
From the method name: 'Escape' as in string escape character (backslash), not the 'Esc' character.

Comment says: IndexOfAny(92, < 32), i.e. 0-31 + '\', which is consistent with the implementation. The space is the "less than" character.

I can re-word the method name/comment to be more clear.

Reply:
> The space is the "less than" character.

I think then using (char)32 rather than Space would be clearer.